### PR TITLE
Prerelease 23.11

### PR DESCRIPTION
**Pull request by t184256**
I give up on doing it cleanly, let's just release already before it's May.

### CODE REVIEW

**Comment by Gerschtli** [comment]
fakedroid seems to be broken.. I will test the new bootstrap on my device, did you push that bootstrap.zip somewhere already?

**Comment by t184256** [comment]
https://nix-on-droid.unboiled.info/bootstrap-prerelease-23.11

https://github.com/nix-community/nix-on-droid/actions/runs/8129783255/job/22217267977 suggests home-manager is broken =/

**Comment by t184256** [comment]
6bccf0a passes on-device and fails on NixOS+fakedroid. so far no idea why shm would be broken in fakedroid, as the entire /dev/ is mounted over

**Comment by Gerschtli** [comment]
The issue with the missing profile.lock is caused by https://github.com/NixOS/nix/pull/5226. A workaround is 

```patch
diff --git c/pkgs/bootstrap.nix i/pkgs/bootstrap.nix
index 7e76299..f7ea097 100644
--- c/pkgs/bootstrap.nix
+++ i/pkgs/bootstrap.nix
@@ -3,7 +3,8 @@
 { runCommand, nixDirectory, prootTermux, bash, pkgs, config, initialPackageInfo }:

 runCommand "bootstrap" { } ''
-  mkdir --parents $out/{.l2s,bin,dev/shm,etc,nix,root,tmp,usr/{bin,lib}}
+  mkdir --parents $out/{.l2s,bin,dev/shm,etc,root,tmp,usr/{bin,lib}}
+  mkdir --parents $out/nix/var/nix/{profiles,gcroots}/per-user/nix-on-droid

   cp --recursive ${nixDirectory}/store $out/nix/store
   cp --recursive ${nixDirectory}/var $out/nix/var
```

**Comment by t184256** [comment]
shm issue seems to be an interplay with link2symlink, as the shm file gets linked around across different names. I guess I'm gonna drop link2symlink in fakedroid

**Comment by t184256** [comment]
Incorporated both changes, let's see what the CI thinks...

**Comment by t184256** [comment]
whoa, CI sure is slow today

**Comment by t184256** [comment]
OK, please review.

If the CI keeps being that unreliable, I think I'll configure a $1 VPS as a self-hosted runner or something. =/

**Review by Gerschtli — APPROVED** [review]
I tested it on my device, works like a charm, thank you!

File: nix-on-droid/nix-on-droid.sh [hunk 1]
```diff
@@ -97,10 +97,16 @@ function doOnDeviceTest() {
     fi
 
     exec "$(nix-instantiate --eval --expr \
-                            "<nix-on-droid/tests/on-device/.run.sh>")"
+                            "<nix-on-droid/tests/on-device/.run.sh>")" "$@"
 }
 
 function doSwitch() {
+    if [[ -e "$HOME/.config/nix-on-droid/flake.nix" && -z "${FLAKE_CONFIG_URI}" ]]; then
```
[the remark this answers is not in the document]
Comment (t184256) [reply nix-on-droid/nix-on-droid.sh:104]: That'd change the default for those who have both configs, I guess.

I'm OK with the default being non-flake while flakes are experimental. We can sure flip it when the bright flake future (c)(tm) finally comes.

**Comment by t184256** [comment]
Thank you for the review and testing!

**Pull request merged**